hmy/api_backend.go

```diff
@@ -321,11 +327,20 @@ var (
 	two = big.NewInt(2)
 )
 
-// GetMedianRawStakeSnapshot  ..
+// GetMedianRawStakeSnapshot ..
 func (b *APIBackend) GetMedianRawStakeSnapshot() *big.Int {
+	b.MedianStakeCache.Mut.Lock()
```

Comment:
No need to later have the unlock scattered all over, just immediately after grabbing the lock, do defer b.MedianStakeCache.Unlock()